launchflow/aws/api_gateway.py

```diff
@@ -60,4 +66,158 @@ def inputs(self, environment_state: EnvironmentState) -> APIGatewayInputs:
         return APIGatewayInputs(
             resource_id=self.resource_id,
             protocol_type=self.protocol_type,
+            cors=self.cors,
+        )
+
+    # TODO: add builder pattern API for adding routes. Requires Resource.subresources()
+    def add_route(self, path: str) -> None:
+        """TODO"""
+        raise NotImplementedError
+
+
+@dataclass
+class APIGatewayLambdaIntegrationInputs(ResourceInputs):
+    api_gateway_id: str
+    function_arn: str
+    function_alias: str
+
+
+@dataclass
+class APIGatewayLambdaIntegrationOutputs(Outputs):
+    api_integration_id: str
+
+
+class APIGatewayLambdaIntegration(AWSResource[APIGatewayLambdaIntegrationOutputs]):
+    """An API Gateway Integration
+
+    ****Example usage:****
+    ```python
+    import launchflow as lf
+
+    api_gateway = lf.aws.APIGateway("my-api-gateway")
+    function = lf.aws.LambdaFunction("my-lambda-function")
+    integration = lf.aws.APIGatewayLambdaIntegrationOutputs(
+        "my-api-gateway-route",
+        api_gateway=api_gateway,
+        function=function,
+    )
+    ```
+    """
+
+    product = ResourceProduct.AWS_API_GATEWAY_LAMBDA_INTEGRATION.value
+
+    def __init__(
+        self,
+        name: str,
+        *,
+        api_gateway: APIGateway,
+        # NOTE: This only supports Lambda function integrations for now
+        function: LambdaFunction,
+    ) -> None:
+        """TODO"""
+        super().__init__(
+            name=name,
+            resource_id=f"{name}-{lf.project}-{lf.environment}",
+        )
+        self._api_gateway = api_gateway
+        self._function = function
+        self.depends_on(
+            api_gateway, function
+        )  # TODO: Remove this once we fix lf.node.Depends
+
+    def inputs(
+        self, environment_state: EnvironmentState
+    ) -> APIGatewayLambdaIntegrationInputs:
+        """Get the inputs required for the API Gateway Lambda Integration.
```

Comment:
nit: I would just remove the doc comment here I don't think it's useful to have it in the docs imo

Reply:
removed all inputs() doc strings